### PR TITLE
Uncommon ASRS monkey/dog/cat stowaway random event

### PR DESCRIPTION
**Pull request by PhantomEpicness**
# About the pull request
Random stowaway monkey/dog/cat/bat event. A small event to break up the boringness of req. When triggered, it will spawn between 1-2 creatures of:

- Monkey and Monkey subtypes
- Cat/Dog
- Bat

# Explain why it's good for the game
Breaking up shipside monotony.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

https://user-images.githubusercontent.com/42422230/217079259-e143077c-2bc2-4f83-8c02-37c86e980acd.mp4

# Changelog

:cl: Totalepicness5
add: Added an Uncommon ASRS monkey/dog/cat stowaway random event
/:cl:

### CODE REVIEW

File: code/game/supplyshuttle.dm [hunk 1]
```diff
@@ -1233,6 +1238,35 @@ var/datum/controller/supply/supply_controller = new()
...
+	handle_animal_event = TRUE
+
+/datum/controller/supply/proc/handle_animal_event()
+	var/monkey_mobs = list(/mob/living/carbon/human/monkey, /mob/living/carbon/human/farwa, /mob/living/carbon/human/stok, /mob/living/carbon/human/yiren, /mob/living/carbon/human/neaera)
+	var/other_animals = list(/mob/living/simple_animal/bat,/mob/living/simple_animal/mouse)
+	var/choice = pick(monkey_mobs,other_animals, "special")
+	var/mob/living/animal = pick(choice)
+	to_chat(world,"run spawn")
+	if(choice == "special")
+		to_chat(world,"spec spawntick")
```
Comment (harryob) [inline code/game/supplyshuttle.dm:1253]: remove this...

File: code/game/supplyshuttle.dm [hunk 2]
```diff
@@ -1233,6 +1238,35 @@ var/datum/controller/supply/supply_controller = new()
...
+		animal = pick(/mob/living/simple_animal/cat,/mob/living/simple_animal/corgi)
+	else
+		for(var/i=0,i < rand(1,2),i++)
+			to_chat(world,"spawntick")
```
Comment (harryob) [inline code/game/supplyshuttle.dm:1257]: ¬¬¬¬

File: code/modules/events/asrs_animals.dm [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	endWhen = 15
+
+/datum/round_event/asrs_animal/setup()
+	. = ..()
+	var/msg
+	msg = "An ASRS animal event has been triggered!"
+	log_admin(msg)
+	message_admins(msg)
+	supply_controller.init_animal_event()
+	to_chat(world,"ebent start")
```
Comment (harryob) [inline code/modules/events/asrs_animals.dm:23]: dasdwad

File: code/modules/events/asrs_animals.dm [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+/datum/round_event/asrs_animal/announce()
+	var/input
+	input = "Unidentified lifesigns detected inside Automated Supply Retrieval System."
+	ai_announcement(input)
```
Comment (harryob) [inline code/modules/events/asrs_animals.dm:28]:
```suggestion
	ai_announcement("Unidentified lifesigns detected inside Automated Supply Retrieval System.")
```

File: code/game/supplyshuttle.dm [hunk 5]
```diff
@@ -1233,6 +1238,35 @@ var/datum/controller/supply/supply_controller = new()
 	/// For code readability.
 	addtimer(CALLBACK(GLOBAL_PROC, /proc/playsound, get_rand_sound_tile(), sound_to_play, 25, FALSE), timer)
 
+
+/datum/controller/supply/proc/init_animal_event()
+	to_chat(world,"init")
+	handle_animal_event = TRUE
+
+/datum/controller/supply/proc/handle_animal_event()
```
Comment (harryob) [inline code/game/supplyshuttle.dm:1246]: ccccodeddddocc....

File: code/game/supplyshuttle.dm [hunk 6]
```diff
@@ -1233,6 +1238,35 @@ var/datum/controller/supply/supply_controller = new()
...
+	var/monkey_mobs = list(/mob/living/carbon/human/monkey, /mob/living/carbon/human/farwa, /mob/living/carbon/human/stok, /mob/living/carbon/human/yiren, /mob/living/carbon/human/neaera)
+	var/other_animals = list(/mob/living/simple_animal/bat,/mob/living/simple_animal/mouse)
+	var/choice = pick(monkey_mobs,other_animals, "special")
+	var/mob/living/animal = pick(choice)
+	to_chat(world,"run spawn")
+	if(choice == "special")
+		to_chat(world,"spec spawntick")
+		animal = pick(/mob/living/simple_animal/cat,/mob/living/simple_animal/corgi)
+	else
+		for(var/i=0,i < rand(1,2),i++)
```
Comment (harryob) [inline code/game/supplyshuttle.dm:1256]:
```suggestion
		for(var/i in rand(1,2))
```

File: code/modules/events/asrs_animals.dm [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+/datum/round_event/asrs_animal
+	announce_when = 1
+	startWhen = 10
+	endWhen = 15
+
+/datum/round_event/asrs_animal/setup()
+	. = ..()
+	var/msg
+	msg = "An ASRS animal event has been triggered!"
```
Comment (harryob) [inline code/modules/events/asrs_animals.dm:19]:
```suggestion
	var/msg = "An ASRS animal event has been triggered!"
```

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/modules/events/asrs_animals.dm (hunk 3 above)
Comment (PhantomEpicness) [reply code/modules/events/asrs_animals.dm:23]: whops sorry

File: code/game/supplyshuttle.dm [hunk 8]
```diff
@@ -1233,6 +1238,35 @@ var/datum/controller/supply/supply_controller = new()
 	/// For code readability.
 	addtimer(CALLBACK(GLOBAL_PROC, /proc/playsound, get_rand_sound_tile(), sound_to_play, 25, FALSE), timer)
 
+
+/datum/controller/supply/proc/init_animal_event()
```
[the remark this answers is not in the document]
Comment (PhantomEpicness) [reply code/game/supplyshuttle.dm:1242]: so true thanks

**Pull request closed without merging**